client/document.go

```diff
@@ -300,7 +306,7 @@ func (doc *Document) setAndParseType(field string, value any) error {
 		}
 
 	// string, bool, and more
-	case string, bool, []any:
+	case string, bool, int64, []any, []bool, []*bool, []int64, []*int64, []float64, []*float64, []string, []*string:
```

Comment:
todo: Please add a separate case for int64. Note that the type of val here will be `any`.

Reply:
Nevermind.  I think it makes no difference since it setCBOR takes an `any`